components/dialog/test/dialog-ifrau.visual-diff.html

```diff
@@ -1,58 +1,65 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
-	<link rel="stylesheet" href="../../../test/styles.css" type="text/css">
-	<script>
-		const preferNative = (window.location.search.indexOf('preferNative=false') === -1);
-		window.D2L = { DialogMixin: { preferNative: preferNative } };
-	</script>
-	<script type="module">
-		import '../../typography/typography.js';
-		import '../../button/button.js';
-		import '../dialog.js';
-	</script>
 	<title>d2l-dialog-ifrau</title>
 	<meta name="viewport" content="width=device-width, initial-scale=1.0">
 	<meta charset="UTF-8">
 </head>
-<body class="d2l-typography">
-	<d2l-button id="open">Open Dialog</d2l-button>
-
-	<d2l-dialog id="ifrau-dialog" title-text="Title" width="400">
-		<div>Line 1</div>
-		<div>Line 2</div>
-		<div>Line 3</div>
-		<div>Line 4</div>
-		<d2l-button slot="footer" primary>Yes</d2l-button>
-		<d2l-button slot="footer" id="cancel">No</d2l-button>
-	</d2l-dialog>
-
-	<script>
+<body>
+	<div id="ifrau-dialog-container"></div>
+
+	<script type="module">
+		import { Host } from 'ifrau/host.js';
+
 		window.ifrauAvailableHeight;
 		window.ifrauTop;
 
-		window.ifrauclient = () => {
-			return {
-				connect: async() => {
-					return {
-						getService: async(name) => {
-							if (name !== 'dialogWC') return null;
-							// mock dialogWC service
-							return {
-								showBackdrop: async() => {
-									return {
-										availableHeight: window.ifrauAvailableHeight,
-										top: window.ifrauTop
-									};
-								},
-								hideBackdrop: async() => {
-								}
-							};
-						}
-					};
-				}
-			};
+		const iframeSrc = decodeURIComponent(window.location.search.split('=')[1]);
+		const iframeOrigin = new URL(iframeSrc).origin;
+		const getContainer = () => document.getElementById('ifrau-dialog-container');
+
+		const verifyPostMessage = (evt, expectedData) => {
+			return evt && evt.data === expectedData && iframeOrigin === evt.origin;
 		};
+
+		const handleIsFramedRequest = evt => {
+			if (!verifyPostMessage(evt, 'isFramedRequest')) return;
+			evt.source.postMessage({ isFramed: true }, iframeSrc);
+		};
+
+		window.addEventListener('message', handleIsFramedRequest, false);
+
+		const marshalDialogOpenEvent = evt => {
+			if (!verifyPostMessage(evt, 'ifrau-dialog-open')) return;
+			getContainer().dispatchEvent(new CustomEvent('ifrau-dialog-open'));
+		};
+
+		window.addEventListener('message', marshalDialogOpenEvent, false);
```

Comment:
I have no idea if marshaling the events this way is actually the easiest/best option.